client/src/components/App.js

```diff
@@ -33,38 +31,34 @@ class App extends Component {
   async showProjects() {
     const response = await fetch('/projects');
     let projects = await response.json();
-    let new_projects = projects.filter(project => project.status === 'new');
-    let in_progress_projects = projects.filter(project => project.status === 'in-progress');
-    let done_projects = projects.filter(project => project.status === 'done');
 
-    this.setState({new_projects, in_progress_projects, done_projects});
+    this.setState({projects});
   }
 
   render() {
+    let project_collections = new Map();
+    this.state.projects.forEach(project => {
+      if (project_collections[project.status]) {
+        project_collections[project.status].push(project);
+      } else {
+        project_collections[project.status] = [project];
+      }
+    });
+    let columns = [];
+    for (let projects_status in project_collections) {
```

Comment:
I feel like this could use the `.map` function, too, maybe… Something like
```javascript
let columns = project_collections.entries().map((title, projects) => <Column projects={projects} title={title} />);
```
(This would also let us have columns with no projects in them, which I think might be helpful later.)

Reply:
I was thinking about "columns with no projects" and for that we'll need to change the db, add a columns table

Reply:
In the meantime, this should still work, since there won't be any empty projects collections. 🙂